viewing_party/party.py

```diff
@@ -1,23 +1,223 @@
+    update_watched = user_data
+    for movie in user_data["watchlist"]:
+        if movie_title in movie["title"]:
+            update_watched["watched"].append(movie)
+            update_watched["watchlist"].remove(movie)
+    print(update_watched)
+    return update_watched    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    
+    users_data = user_data
+    avg_values = 0
+    counter = 0
+    movies = users_data["watched"]
+    if movies:
+        for items in users_data.values():
+            for num in items:
+                avg_values += num["rating"]
+                counter += 1
+        result = avg_values/counter  
+        return result
+    else:
+        return 0
+
+# for items in users_data.values(): is'nt this the same as "movies"?
+# def get_most_watched_genre(user_data):
+#     users_data = user_data
+#     data = user_data.values()
+#     list = []   
+#     movies = users_data["watched"]
+
+#     if movies:
+#         for item in data:
+#             for i in item:
+#                 list.append(i["genre"])
+#         most_commom = max(set(list), key = list.count)
+#         return most_commom  
+#     else: 
+#         return most_commom
+
+
+def get_most_watched_genre(user_data):
+
+    watched_genres = {}
+    watched_movies = user_data["watched"]
+
+    if not watched_movies:
+        return None
+
+    for movie in watched_movies:
+        if movie["genre"] in watched_genres:
+            watched_genres[movie["genre"]] += 1
+        else: 
+            watched_genres[movie["genre"]] = 1
+    list_of_freq = list(watched_genres.values())
+    list_of_freq.sort()
+    most_watched_value = list_of_freq[-1]
+
+    for genre in watched_genres:
+        if watched_genres[genre] == most_watched_value:
+            most_watched_genre = genre
+    return most_watched_genre
+
+def friends_watched(user_data):
+
+    users_data = user_data
+    my_movies = users_data["watched"]
+    friends = users_data["friends"]
+    friends_watched_movies = []
+    if my_movies:
+        for movies in friends:
+            for movie in movies["watched"]:
+                if movie not in friends_watched_movies:
+                    friends_watched_movies.append(movie)
+        return friends_watched_movies
+    else:
+        return friends_watched_movies  
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    users_data = user_data
+    my_movies = users_data["watched"]
+    friends_watched_movies = friends_watched(user_data)
+    unique_list = []   
+    
+
+    if my_movies:
+        for movie in my_movies:
+            if movie not in friends_watched_movies:
+                unique_list.append(movie)
+        return unique_list
+    else:
+        return unique_list  

```

Comment:
Since `unique_list` is initialized to an empty list, and then returned at the end of both the `if` and `else` branches, we could drop the `else` and move the return to after the conditional (it gets returned either way). Similar to what you did for `get_friends_unique_watched`.

```python
    if my_movies:
        for movie in my_movies:
            if movie not in friends_watched_movies:
                unique_list.append(movie)

    return unique_list
```

Reply:
Here again, we could think about making a set of titles to help us check for uniqueness more quickly than using `in` on the movie list. This would also be a little safer, since we can't be sure that the movie keys in the friends list will always be the same as the movie keys for the user (recall that in wave 2, the user movies had a rating).